libnavigator/src/androidTest/java/com/mapbox/navigation/navigator/internal/NavigatorTest.kt

```diff
@@ -0,0 +1,123 @@
+package com.mapbox.navigation.navigator.internal
+
+import androidx.annotation.RawRes
+import androidx.test.espresso.Espresso
+import androidx.test.platform.app.InstrumentationRegistry
+import com.mapbox.api.directions.v5.DirectionsCriteria
+import com.mapbox.api.directions.v5.models.DirectionsResponse
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.navigator.internal.util.readRawFileText
+import com.mapbox.navigation.navigator.internal.util.runOnMainSync
+import com.mapbox.navigator.CacheFactory
+import com.mapbox.navigator.CacheHandle
+import com.mapbox.navigator.ConfigFactory
+import com.mapbox.navigator.ConfigHandle
+import com.mapbox.navigator.Navigator
+import com.mapbox.navigator.NavigatorConfig
+import com.mapbox.navigator.ProfileApplication
+import com.mapbox.navigator.ProfilePlatform
+import com.mapbox.navigator.Routes
+import com.mapbox.navigator.SettingsProfile
+import com.mapbox.navigator.TilesConfig
+import org.junit.Assert.assertTrue
+import org.junit.Test
+
+class NavigatorTest {
+
+    private companion object {
+        @RawRes
+        private val DIRECTIONS_RESPONSE =
+            com.mapbox.navigation.navigator.test.R.raw.directions_response_two_leg_route
+    }
+
+    @Test(timeout = 10000)
+    fun successfulSetRoutesToNN() {
+        Espresso.onIdle()
```

Comment:
why do you need to call `onIdle`?

Reply:
`Espresso#onIdle` wait for result in `runOnMainSync` lambda to stop test [doc](https://developer.android.com/reference/android/support/test/espresso/Espresso.html#onIdle())

Reply:
Looks like it waits earlier that `runOnMainSync` get's executed, doesn't it?

Reply:
`runOnMainSync` is blocking, isn't it? why do you need to wait for it?

Reply:
you're probably right. Need to put `Espresso.onIdle()` to the bottom of the test, to idle till callback is invoked